sdks/python/apache_beam/runners/dataflow/dataflow_runner.py

```diff
@@ -126,19 +126,20 @@ def rank_error(msg):
         messages, page_token = runner.dataflow_client.list_messages(
             job_id, page_token=page_token, start_time=last_message_time)
         for m in messages:
-          if last_message_id is not None and m.id == last_message_id:
+          message = '%s: %s: %s' % (m.time, m.messageImportance, m.messageText)
+          m_hash = hash(message)
```

Comment:
No it is not. `hash(m)` ends up being different for different instances of `m` with the same field values.